### PR TITLE
Dp 20050 font switch

### PR DESCRIPTION
**Pull request by clairesunstudio**
This PR changed the typeface of Mayflower from Texta to Noto Sans.
Changed:
  - project: Assets
    component: Fonts
    description: Switch fonts from Texta to Noto Sans, removed fallback fonts and added language support. (#1193)
    issue: DP-20050
    impact: Major
Added:
  - project: Assets
    component: Fonts
    description: Added `$fonts-enable-rtl` global variable to allow setting direction to "right-to-left" for languages like Arabic, Persian, Urdu and Hebrew. (#1193)
    issue: DP-20050
    impact: Major
Added:
  - project: Assets
    component: Fonts
    description: Added `$fonts-langs-support` global variable to conditionally load and render additional fonts for language support. (#1193)
    issue: DP-20050
    impact: Major
Added:
  - project: Assets
    component: Fonts
    description: Added `$fonts-display-global` global variable to control the custom fonts download and render behavior. (#1193)
    issue: DP-20050
    impact: Major

To view the changes to Patternlab:
- [ ] check out this branch and `cd patternlab/styleguide` and `npm link @massds/mayflower-assets`
- [ ] `gulp`
- [ ] http://localhost:3000
---

To view the changes to React go to typography docs page: 
- [ ] check out this branch and `cd react` and `npm link @massds/mayflower-assets`
- [ ] `npm start`
- [ ] http://localhost:6006/?path=/docs/brand-typography--noto-sans

---

Compared with Texta, Noto Sans offer the following benefits:
- An open-source license
- Broader language support (more than 800 languages, covering all the Mass.gov supported languages) removing the need for fallback fonts. e.g.  Khmer tofu and Vietnamese ransom notes and other language layout problems
Screen Shot 2020-10-07 at 10 37 43 AM
Screen Shot 2020-10-07 at 1 24 56 PM

Screen Shot 2020-10-06 at 9 23 35 AM

- Monospaced fonts in the same family of typeface for tabular display
- Bigger x-heights and slimmer character widths

----

- Added variable fonts and static fonts for Noto Sans, using `@support` check to load variable fonts conditionally.
- Added additional necessary Noto Sans fonts for language support, using Unicode range and [glyphhange](https://www.npmjs.com/package/) to create and load font subsets e.g.
```
// variable fonts
@supports (font-variation-settings: normal) {
  @font-face {
    font-family: "Noto Sans Khmer VF";
    src: url("/static/fonts/noto/Khmer/NotoSansKhmer-VF-subset.woff2") format("woff2"), url("/static/fonts/noto/Khmer/NotoSansKhmer-VF-subset.woff") format("woff");
    font-weight: 100 900;
    font-style: normal;
    font-display: "swap";
    unicode-range: U+1780-17FF, U+19E0-19FF; } 
}
// static fallback fonts
@font-face {
  font-family: "Noto Sans Khmer";
  src: url("/static/fonts/noto/Khmer/NotoSansKhmer-Bold-subset.woff2") format("woff2"), url("/static/fonts/noto/Khmer/NotoSansKhmer-Bold-subset.woff") format("woff");
  font-weight: 550;
  font-style: normal;
  font-display: "swap";
  unicode-range: U+1780-17FF, U+19E0-19FF; }
@font-face {
  font-family: "Noto Sans Khmer";
  src: url("/static/fonts/noto/Khmer/NotoSansKhmer-Regular-subset.woff2") format("woff2"), url("/static/fonts/noto/Khmer/NotoSansKhmer-Regular-subset.woff") format("woff");
  font-weight: 400;
  font-style: normal;
  font-display: "swap";
  unicode-range: U+1780-17FF, U+19E0-19FF; }
```

- Also using language codes to conditionally load the font faces.
```
*:lang(km) {
  font-family: "Noto Sans Khmer VF", "Noto Sans Khmer", "Noto Sans VF", "Noto Sans", "Helvetica", "Arial", sans-serif; }
*:lang(Khmr) {
  font-family: "Noto Sans Khmer VF", "Noto Sans Khmer", "Noto Sans VF", "Noto Sans", "Helvetica", "Arial", sans-serif; }
```
Screen Shot 2020-10-07 at 10 35 57 AM

To add additional fonts for language support, follow the Myanmar example --> 018ba24
Before:
Screen Shot 2020-10-06 at 7 42 27 PM
After:
Screen Shot 2020-10-06 at 7 50 11 PM

### CODE REVIEW

**Comment by clairesunstudio** [comment]
Reverted the fix for issue https://github.com/massgov/mayflower/issues/1067 in pattenlab, added variable `enable-rtl` to conditionally allow `direction: rtl` for rtl scripts b63fe5a

Need to come up with design solutions for RTL incompatible components, e.g.
Screen Shot 2020-10-07 at 9 42 45 AM

File: assets/scss/00-base/_fonts-langs.scss [hunk 1]
```diff
@@ -0,0 +1,201 @@
+@forward "variables";
```
Comment (smurrayatwork) [inline assets/scss/00-base/_fonts-langs.scss]:
```suggestion
```

File: assets/scss/00-base/_fonts-langs-import.scss [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+$warang-citi:			U+118A0-118FF;
+$zanabazar:				U+11A00-11A4F;
+$lgc: 						U+0000-00FF, U+0100-017F, U+0180-024F, U+0370-03FF, U+0259-03C0, U+1E00-1EFF, U+2000-FB02, U+1F00-1FFF, U+0400-04FF, U+0500-052F, U+1C80-1C8F;
+
+
+$oneSpace: 	' ';
+$oneDash:		'-';
+$global-font-display:	'swap';
+
+@mixin font-face ($family: "Noto", $type: "Sans", $script: "Cyrillic", $suffix: "UI", $weight: "VF", $subset: "-subset", $fontWeight: "100 900", $fontStyle: "normal", $fontDisplay: "swap", $unicodeRange: "cyrillic", $space: $oneSpace, $dash: $oneDash) {
```
Comment (smurrayatwork) [inline assets/scss/00-base/_fonts-langs-import.scss]: Move this mixin to 00-base/mixins/_font-face.scss and then add a `@use '00-base/mixins/font-face'`;

File: assets/scss/00-base/_fonts-langs.scss [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+@use "variables" as *;
+// Language sets and @mixin
+
+// This adds all of the language sets for a script to an `&:lang()` rule
+// it is used in script selector files so they can easily be applied to each font style
+@mixin langlist($stack, $dir, $langs) {
```
Comment (smurrayatwork) [inline assets/scss/00-base/_fonts-langs.scss]: Move this to the mixins directory and `@use` it in this file.

File: assets/scss/00-base/_fonts-langs.scss (hunk 1 above)
Comment (smurrayatwork) [inline assets/scss/00-base/_fonts-langs.scss]:
```suggestion
```

File: assets/scss/00-base/_fonts-langs-import.scss [hunk 4]
```diff
@@ -0,0 +1,152 @@
+@forward "paths";
```
Comment (smurrayatwork) [inline assets/scss/00-base/_fonts-langs-import.scss]:
```suggestion
```

File: assets/scss/00-base/_fonts-langs-import.scss [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+@forward "variables";
```
Comment (smurrayatwork) [inline assets/scss/00-base/_fonts-langs-import.scss]:
```suggestion
```

File: assets/scss/00-base/_fonts-langs-import.scss [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+@use "paths" as *;
+@use "variables" as *;
+// Variable sans fonts
+$adlam:						U+1E900-1E95F, U+2E41; // Uses reversed comma
```
Comment (smurrayatwork) [inline assets/scss/00-base/_fonts-langs-import.scss]: Make sure to not use any of these variables globally in your `@use` statement (do not use `as *`).

File: assets/scss/00-base/_fonts.scss [hunk 7]
```diff
@@ -6,143 +6,64 @@
 // Black => 800
 
 // Set $assets-path to form the correct links to the parent folder of fonts/ based on the location of the compiled css
-
 @forward "paths";
```
Comment (smurrayatwork) [inline assets/scss/00-base/_fonts.scss]:
```suggestion
```

**Review by smurrayatwork — APPROVED** [review]

**Pull request merged**